### PR TITLE
Optimization release tidb wokflow

### PR DESCRIPTION
**Pull request by dianqihanwangzi**
- push dockerhub and ucloud together, reduce docker pull time.
- parallel run tiup release and tarball job. 
- upgrade go version to 1.16.4(release_tag >v5.1.0)

### CODE REVIEW

File: jenkins/pipelines/cd/optimization-release-tidb.groovy [hunk 1]
```diff
@@ -622,52 +659,111 @@ __EOF__
...
-                                [$class: 'StringParameterValue', name: 'PD_TAG', value: PD_TAG],
-                                [$class: 'StringParameterValue', name: 'BINLOG_TAG', value: BINLOG_TAG],
-                                [$class: 'StringParameterValue', name: 'LIGHTNING_TAG', value: BR_TAG],
-                                [$class: 'StringParameterValue', name: 'BR_TAG', value: BR_TAG],
-                                [$class: 'StringParameterValue', name: 'CDC_TAG', value: CDC_TAG],
-                                [$class: 'StringParameterValue', name: 'TIFLASH_TAG', value: TIFLASH_TAG],
-                                [$class: 'StringParameterValue', name: 'DUMPLING_TAG', value: DUMPLING_TAG],
-                        ]
+            
+            // 这一步显得没有必要
```
Comment (dianqihanwangzi) [inline jenkins/pipelines/cd/optimization-release-tidb.groovy:718]: this step seems useless,v1.0.1 has already pushed to cloud

File: jenkins/pipelines/cd/optimization-release-tidb.groovy [hunk 2]
```diff
@@ -622,52 +659,111 @@ __EOF__
...
+                }
             }
+            // tiflash 上传二进制，制作上传镜像 
+            builds["Push tiflash Docker"] = {
+                build job: 'release_tiflash_by_tag',
+                        wait: true,
+                        parameters: [[$class: 'StringParameterValue', name: 'RELEASE_TAG', value: "${RELEASE_TAG}"]]
 
-            stage("Push tarbll/image") {
-                parallel builds
```
Comment (zhouqiang-cl) [inline jenkins/pipelines/cd/optimization-release-tidb.groovy:634]: Seems we still need parallel builds?

File: jenkins/pipelines/cd/optimization-release-tidb.groovy [hunk 3]
```diff
@@ -622,52 +659,111 @@ __EOF__
...
+
+// monitoring 编译，upload
+            // stage("trigger release monitor") {
+            //     build job: 'release-monitor',
+            //             wait: true,
+            //             parameters: [[$class: 'StringParameterValue', name: 'RELEASE_TAG', value: "${RELEASE_TAG}"]]
+            // }
 
+            stage("Trigger jira version") {
+                build(job: "jira_create_release_version", wait: false, parameters: [string(name: "RELEASE_TAG", value: "${RELEASE_TAG}")])
```
Comment (zhouqiang-cl) [inline jenkins/pipelines/cd/optimization-release-tidb.groovy:741]: I think we should remove `jira_create_release_version`, it is too ugly in CD

**Review by zhouqiang-cl — APPROVED** [review]
rest LGTM

**Pull request merged**